src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs

```diff
@@ -170,6 +171,31 @@ protected virtual void InitializeEnvironmentVariables(Dictionary<string, string>
             {
                 InitializeHttpRequestEnvironmentVariables(environmentVariables, request);
             }
+
+            InitializeIdentityEnvironmentVariables(environmentVariables, context.Identity);
+        }
+
+        internal static void InitializeIdentityEnvironmentVariables(Dictionary<string, string> environmentVariables, ClaimsIdentity identity)
+        {
+            environmentVariables["IDENTITY_AUTHENTICATION_TYPE"] = identity.AuthenticationType;
+
+            foreach (var claim in identity.Claims)
+            {
+                // some script environments have problems with special characters in
```

Comment:
Anyone have any suggestions on alternate ways of handling this? I ran into this issue in my testing - claim names can be very long and contain characters that are special in languages like Batch.